src/util/db/dbid.h

```diff
@@ -24,33 +24,19 @@
 class DbId {
 protected:
 public:
-    // Alias for the corresponding native type. It keeps the
-    // implementation of this class flexible if we ever gonna
-    // need to change it from 'int' to 'long' or any other type.
-    typedef int value_type;
-
-    DbId()
-        : m_value(kInvalidValue) {
-        DEBUG_ASSERT(!isValid());
-    }
-    explicit DbId(value_type value)
-        : m_value(value) {
-        DEBUG_ASSERT(isValid() || (kInvalidValue == m_value));
-    }
-    explicit DbId(QVariant variant)
-        : DbId(valueOf(std::move(variant))) {
-    }
+  constexpr DbId()
+          : m_value(kInvalidValue) {
+  }
+  explicit DbId(const QVariant& variant)
+          : m_value(valueOf(variant)) {
+  }
```

Comment:
since you eliminated the `int` from the interface, how can we ensure to avoid the QVariant construction overhead in the case where we are getting passed an int?

Reply:
We always get a QVariant.

Reply:
What is the conclusion here?

Reply:
well, I'd either like to see a constructor taking `int` (at which point the `int` is part of the interface again), or a deleted ctor taking `int` to warn us of the fact that we're passing an int directly... Anyways, I'm probably overcomplicating the issue as always. resolved.

Reply:
Deleting `DbId(int)` was a good idea. It reveals some int roundtrips. I have removed them from the Mixxx code. 
Unfortunately we have some constants that are derived from `int` the testing code. I have fixed them in the latest commit.